feat: Temporary placeholders for unfinished pages on doc website

elephantcatdog opened the pull request:
## Changes
I thought it would be helpful to have placeholders for unfinished pages on the doc website so that people going here or being redirected here from ux.bentley.com aren't only getting a bunch of mostly empty pages with little information on them.

I want to add it to all the unfinished pages, but wanted to see if this format/wording is good before copy/pasting to all the other pages. I added the lines before and after so that it's a bit more noticeable. Also I'm not sure if it's better to have the full link be visible vs just having 'storybook' with a link.

![image](https://user-images.githubusercontent.com/37936169/221315847-c547ee87-c9cb-47b8-8b99-0049b92b5667.png)

## Testing
N/A

## Docs
N/A

gretanausedaite reviewed (commented):
Very good idea!
Maybe we could add some kind of banner/alert instead of simple text? You could create a component and pass down all mdx pages that need this info. Also, you could create new styling for this component, it doesn't have to be iTwinUI - you can use our theme gradients or something.

mayank99 commented:
Agreed with @gretanausedaite

elephantcatdog commented:
I'm not sure why, but the grid template isn't working correctly for the alert I added: 
![image](https://user-images.githubusercontent.com/37936169/221958179-8dc18440-8b1e-4480-a6b2-08bcbe1b1094.png)

This was supposed to be a quick addition, so I didn't make the alert look fancy or anything.

gretanausedaite reviewed (commented):
Grid does not work because Alert does not get "placeholder" class. 
Add classname prop (eg. Card.astro) or move styles to Placeholder.astro.
Also I think it'd look better if alert had space before title (bottom padding) (eg. var(--space-5)). 

You sure you don't want to play around with styles? 
![image](https://user-images.githubusercontent.com/83585998/222156898-04b842e7-57f7-42ca-9607-97dfea27c1f4.png)

gretanausedaite commented on apps/website/src/components/Placeholder.astro:
```diff
@@ -0,0 +1,19 @@
+  <span class='iui-alert-message'
+    >Some pages have not yet been finished and are being worked on. In the meantime, visit <a
+      class='iui-alert-link'
+      href='https://itwin.github.io/iTwinUI/react/?path=/story/overview--page'>storybook</a
```
It would be cool if we routed users to same component in storybook. Can be added later.

elephantcatdog replied:
Updated

mayank99 commented:
I think it might be better to control this message at a page-level rather than adding to the grid on all pages.

This will also allow customizing the message/link for each page.

The implementation could look like one of these two options:

 - A simple component that takes the component name (or url) as prop.
   ```mdx
   ---
   title: Alert
   ...
   ---
   ...
   <PlaceholderMessage component='alert' />
   ```

 - Alternatively, use a frontmatter property to indicate that the page should have a message.
   ```mdx
   ---
   title: Alert
   ...
   wip: true
   ---
   ...
   ```
   After this, you'll be able to access this property (among other properties such as `title`) in the [layout](https://docs.astro.build/en/core-concepts/layouts/#markdownmdx-layouts) and decide what to show.

elephantcatdog commented:
@mayank @gretanausedaite 

Thank you both for your input! I've updated the placeholder. Now it's got a property for the end of the storybook url, and is placed on an individual component's page vs the layout. I also updated the visuals to something similar to what you showed, Greta, because I thought that looked nice.

If this looks good, then I'll put the placeholder on all the pages that need it. I have it on 'Avatar' and 'Carousel' right now (and will remove it from 'Avatar').

![image](https://user-images.githubusercontent.com/37936169/223254089-289bd748-26f2-4c0b-be86-99be35be0608.png)

mayank99 commented on apps/website/src/components/Placeholder.astro:
```diff
@@ -0,0 +1,39 @@
+  componentPageName: string;
+};
+
+const { componentPageName } = Astro.props;
```
you might want to take the rest props and add them to the root element
```
const { componentPageName, ...rest } = Astro.props;
---
<div {...rest}>
```

elephantcatdog replied:
Updated

mayank99 commented on apps/website/src/components/Placeholder.astro:
```diff
@@ -0,0 +1,39 @@
+</div>
+
+<style lang='scss'>
+  .alert {
```
styles are automatically scoped to the component, so you can just use `div` here instead of `.alert`

mayank99 commented on apps/website/src/components/Placeholder.astro:
```diff
@@ -0,0 +1,39 @@
+
+<div class='alert' data-iui-status='informational'>
+  <SvgInfoCircular className='iui-alert-icon' aria-hidden='true' />
+  <span class='iui-alert-message'
```
we are trying to maintain a distinct design language for the website, so let's not use anything from iTwinUI (no `iui-` classes / itwinui-icons).

you don't need to make anything fancy, but would be nice if you can use spacing/color values from `__global.astro` (line 116 onwards).

for icons, you can grab something from https://icones.js.org/collection/all and either add it to the `website/src/icons/` folder or inline it in CSS.

mayank99 replied:
here is an example implementation you can build upon:
```astro
---
export type Props = {
  /** ending of the url for a specific storybook page */
  componentPageName: string;
};

const { componentPageName, ...rest } = Astro.props;

const storyUrl = `https://itwin.github.io/iTwinUI/react/?path=/story/${componentPageName}`;
---

<div {...rest}>
  <p>
    This page has not yet been finished and is being worked on. In the meantime, please visit
    <a href={storyUrl}>storybook</a> to see more examples.
  </p>
</div>

<style lang='scss'>
  div {
    border-block: 1px solid var(--color-line-1);
    padding-block: var(--space-3);
    display: flex;
    align-items: center;
    gap: var(--space-2);

    &::before {
      content: '';
      flex-shrink: 0;
      display: inline-block;
      width: 1.5em;
      height: 1.5em;

      --_svg: url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" width="32" height="32" viewBox="0 0 24 24"><path fill="currentColor" d="M13 9a1 1 0 1 1-2 0a1 1 0 0 1 2 0Zm-1 2.75a.75.75 0 0 1 .75.75v5a.75.75 0 1 1-1.5 0v-5a.75.75 0 0 1 .75-.75Z"/><path fill="currentColor" fill-rule="evenodd" d="M14.27 3.993a2.749 2.749 0 0 0-4.54 0l-.432.632a75.95 75.95 0 0 0-6.944 12.563l-.09.208a2.511 2.511 0 0 0 2.024 3.497a69.43 69.43 0 0 0 15.424 0a2.511 2.511 0 0 0 2.024-3.497l-.09-.208a75.951 75.951 0 0 0-6.944-12.563l-.432-.632Zm-3.302.846a1.25 1.25 0 0 1 2.064 0l.432.632a74.444 74.444 0 0 1 6.806 12.315l.09.208a1.011 1.011 0 0 1-.814 1.408c-5.015.56-10.077.56-15.092 0a1.011 1.011 0 0 1-.815-1.408l.09-.208a74.45 74.45 0 0 1 6.807-12.315l.432-.632Z" clip-rule="evenodd"/></svg>');
      -webkit-mask-image: var(--_svg);
      mask: var(--_svg) no-repeat center / contain;
      background-color: currentColor;
    }
  }
</style>

```

elephantcatdog replied:
Updated

mayank99 reviewed (commented):
usage looks good so i think it can be added to all pages as-is, but i think the styling could be toned down to be less distracting.

elephantcatdog commented:
![image](https://user-images.githubusercontent.com/37936169/223490182-29abfa92-d0ef-4192-a1b2-108ba2365c56.png)

mayank99 commented on apps/website/src/components/Placeholder.astro:
```diff
@@ -0,0 +1,37 @@
+    border-radius: var(--border-radius-1);
+
+    svg {
+      color: #adb7f5;
```
can you use one of the color variables defined in `_global.astro` (or add a new one if you think it makes sense)?

elephantcatdog replied:
Updated

veekeys commented:
> Updated look: ![image](https://user-images.githubusercontent.com/37936169/223490182-29abfa92-d0ef-4192-a1b2-108ba2365c56.png)

Would it be possible to get rid of any CONNECT related references in public docs site? I do understand initially we might have copied some of the old content, but in this particular case (hopefully not that many in all docs site) it is something we do not want to even mention to anyone anymore.

mayank99 commented:
> Would it be possible to get rid of any CONNECT related references in public docs site? I do understand initially we might have copied some of the old content, but in this particular case (hopefully not that many in all docs site) it is something we do not want to even mention to anyone anymore.

Agreed we should remove that.

Almost every single page had content copied over from the old docs, so there might be more weird stuff. #961 is tracking the status of pages that have been further reviewed/adjusted after the old content was moved.

FlyersPh9 reviewed (approved):
Once [this comment](https://github.com/iTwin/iTwinUI/pull/1099/files#r1128205630) is resolved, looks good to me!

elephantcatdog commented:
Okay, I added the placeholder to all the unfinished pages, and confirmed that the links go where they're supposed to. I have a question about one of the pages though - 'Typography' is a page that has 'text' and 'label' on it, but we have separate pages for the other types of typography. Why does this page exist as it is, and not either include all types of typography or have separate pages for all?

gretanausedaite commented:
> I have a question about one of the pages though - 'Typography' is a page that has 'text' and 'label' on it, but we have separate pages for the other types of typography. Why does this page exist as it is, and not either include all types of typography or have separate pages for all?

What are "other types of typography"?

elephantcatdog commented:
> What are "other types of typography"?

These are all listed under "Typography" in storybook:
![image](https://user-images.githubusercontent.com/37936169/223749771-1d8c826e-8dd3-4c87-97fe-876fc013bbb8.png)

gretanausedaite commented:
> These are all listed under "Typography" in storybook: ![image](https://user-images.githubusercontent.com/37936169/223749771-1d8c826e-8dd3-4c87-97fe-876fc013bbb8.png)

They are under "typography" in side nav.
![image](https://user-images.githubusercontent.com/83585998/223768880-47c48855-3db9-4c65-9361-2bda2abe4eff.png)

We've just combined text and label because it would be hard to separate them visually in all components page
![image](https://user-images.githubusercontent.com/83585998/223769038-b4ea7c9a-53e9-4fbf-b50e-623794d4e15a.png)

elephantcatdog commented:
Hm, I see, thanks Greta
